inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/source/reader/InlongIcebergSourceReaderMetrics.java

```diff
@@ -48,8 +52,29 @@ public void registerMetrics(MetricOption metricOption) {
 
     public void outputMetricsWithEstimate(ArrayBatchRecords batchRecord) {
         if (sourceMetricData != null) {
-            sourceMetricData.outputMetricsWithEstimate(batchRecord.records());
+            int dataCount = batchRecord.numberOfRecords();
+            Object[] records = batchRecord.records();
+            for (int i = 0; i < dataCount; i++) {
+                long dataSize = getDataSize(records[i]);
+                long dataTime = getDataTime(records[i]);
+                sourceMetricData.outputMetrics(1, dataSize, dataTime);
+            }
+
         }
+    }
 
+    private long getDataTime(Object object) {
```

Comment:
maybe change to Object data to make more sense

Reply:
Fixed, thx